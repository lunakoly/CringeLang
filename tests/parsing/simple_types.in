var a = 10
var b = "Hello!"

if a + b
    c = a + b
    let d = b
else
    "sorry"

fun go
    let NAME = "Nick"
    NAME

go.NAME

var test: Int -> Int
let fest: Bool -> Bool
typealias Callback = String -> String
typealias Callback = Char
let rest: Char
let mest: Fhar
