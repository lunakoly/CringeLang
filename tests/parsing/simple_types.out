==== Diagnostics ====

==== AST ====
[var [a]: <!MISSING TYPE!><!> = [10], var [b]: <!MISSING TYPE!><!> = ["Hello!"], if (a + b) [[([c] = [(a + b)]), let [d]: <!MISSING TYPE!><!> = [b]]] else [[["sorry"]]], fun go ([]): <!MISSING RETURN TYPE!><!> [[let [NAME]: <!MISSING TYPE!><!> = ["Nick"], [NAME]]], [go.NAME], var [test]: (Int -> Int), let [fest]: (Bool -> Bool), typealias Callback = (String -> String), typealias Callback = Char, let [rest]: Char, let [mest]: Fhar]

==== RESOLVED ====
*** FILE 0 ***
[var [a]: Int = [10], var [b]: String = ["Hello!"], if (a + b) [[([c] = [(a + b)]), let [d]: String = [b]]] else [[["sorry"]]], fun go ([]): String [[let [NAME]: String = ["Nick"], [NAME]]], [go.NAME], var [test]: [BINARY], let [fest]: [BINARY], typealias Callback = [BINARY], typealias Callback = Char, let [rest]: Char, let [mest]: [ERROR:UNRESOLVED_IDENTIFIER]]

==== Done ====
